lib/common/formats.js

```diff
@@ -347,5 +347,69 @@ module.exports = {
       })
       .value();
     return JSON.stringify(to_ret, null, 2);
+  },
+
+  /**
+   * Creates a flatten JSON from input, capitalizing and then concatenating 
+   * each key with it's pre-key recursively.
+   *
+   * @memberof Formats
+   * @kind member
+   * @example
+   * ```json
+    {
+      "size": {
+        "font": {
+           "small": { "value": "13px" },
+           "medium": { "value": "15px" },
+           "large": { "value": "17px" },
+           "base": { "value": "{size.font.medium.value}" }
+        }
+      }
+    }```
+   * 
+   * Creates:
+   * ```json
+    {
+      "SizeFontSmall": "13px",
+      "SizeFontMedium": "15px",
+      "SizeFontLarge": "17px",
+      "SizeFontBase": "15px"
+    }
+   * ```
+   */
+  'json/flat': function(dictionary) {
+    var evaluateProp = function(property) {
+      // property value is in format '{*}'
+      var isReference = /^\{.*\}/g.test(property);
+      if (isReference) {
+        // removing brackets, taking 'content' out of '{content}'
+        property = property.substr(1, property.length - 2);
+        return _.get(dictionary, property);
+      }
+      return property;
+    };
+
+    // recursive, initial prefix is none (empty string)
+    // iterate over keys capitalizing and concatenating them 
+    // until "value" key is found, then evaluate it's value
+    var deepFlatten = function(object, prefix) {
```

Comment:
probably I am missing something, but I think that at this point in the pipeline you already have the `prop.name` and the `prop.value` (and potentially also the `property.path`), no? can you compare with other formats and see if they may be of help?

Reply:
Didn't get it. Where I'd have `prop.name` or `prop.path`? There's another pre-processor for JSON that can do this?
In this code I'm iterating recursively (like a DFS) and also saving the actual path until now to build the final path/name for each property

Reply:
@didoo is correct, see my comment above about Style Dictionary already processing the input JSON. The formatter function takes 2 arguments, a `dictionary` and a `config`. Both are objects, the important one for this is the `dictionary` which has 2 attributes: `properties` and `allProperties`. `allProperties` is already a flattened list of the design tokens (properties) which have already been processed. If you take a look at the 'javascript/es6' format you can see this in action.